### PR TITLE
Combine plugins and integration notes under Integrations

### PR DESCRIPTION
**Pull request by tpdorsey**
Combines plugin and integration pages under a single Integrations tab.

please review

## Change Summary

* Renames Plugins nav tab to Integrations
* Moves integration topics to Integrations
* Combines mypy integration notes under mypy plugin topic
* Removes mypy integration page with redirect from old URL
* Adds mkdocs-redirects plugin to support page redirects

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @samuelcolvin

### CODE REVIEW

File: docs/mypy_plugin.md
Comment (samuelcolvin) [inline docs/mypy_plugin.md:1]: we probably need to rename this file to `mypy`.

We also need to keep a list of redirects we'll need to add before the v2 release.

File: mkdocs.yml [hunk 1]
```diff
@@ -56,15 +56,14 @@ nav:
   - usage/dataclasses.md
   - usage/validation_decorator.md
   - usage/postponed_annotations.md
-  - 'Usage with mypy': usage/mypy.md
-  - 'Usage with devtools': usage/devtools.md
-  - 'Usage with rich': usage/rich.md
-- Plugins:
+- Integrations:
   - 'Mypy plugin': mypy_plugin.md
```
Comment (samuelcolvin) [inline mkdocs.yml:60]: rename to `Mypy`

File: mkdocs.yml [hunk 2]
```diff
@@ -56,15 +56,14 @@ nav:
...
   - 'PyCharm plugin': pycharm_plugin.md
-  - 'Visual Studio Code': visual_studio_code.md
   - 'Hypothesis plugin': hypothesis_plugin.md
+  - 'Visual Studio Code': visual_studio_code.md
   - 'Code Generation': datamodel_code_generator.md
+  - 'Using Pydantic with devtools': usage/devtools.md
```
Comment (samuelcolvin) [inline mkdocs.yml:65]: we should probably rename these files.

File: mkdocs.yml [hunk 3]
```diff
@@ -96,3 +95,6 @@ plugins:
       on_pre_build: 'docs.plugins.main:on_pre_build'
       on_files: 'docs.plugins.main:on_files'
       on_page_markdown: 'docs.plugins.main:on_page_markdown'
+- redirects:
+    redirect_maps:
+      'usage/mypy.md': 'mypy_plugin.md'
```
Comment (samuelcolvin) [inline mkdocs.yml:100]: the lack of newline here is causing linting to fail.

Good catch on redirects :+1:

File: docs/mypy_plugin.md [hunk 4]
```diff
@@ -155,3 +155,33 @@ init_forbid_extra = true
 init_typed = true
 warn_required_dynamic_aliases = true
 ```
+## Using mypy without the plugin
```
Comment (samuelcolvin) [inline docs/mypy_plugin.md:158]: I think we want this section before starting on the plugin.

**Review by samuelcolvin — COMMENTED** [review]
otherwise looking good.

Please update.

**Comment by tpdorsey** [comment]
Ready for another review

**Comment by samuelcolvin** [comment]
tests and linting are failing, it might be worth using pre-commit. Please update.

**Pull request merged**